joplin/settings.py

```diff
@@ -449,9 +445,13 @@ def show_toolbar(request):
 JOPLIN_APP_HOST_PORT = os.getenv('JOPLIN_APP_HOST_PORT', 8000)
 # The CMS_API endpoint of the current Django App for published Janis to use
 if IS_LOCAL or IS_TEST:
-    CMS_API = f"http://localhost:{JOPLIN_APP_HOST_PORT}/api/graphql"
+    # Base URL to use when referring to full URLs within the Wagtail admin backend -
+    # e.g. in notification emails. Don't include '/admin' or a trailing slash
+    BASE_URL = f'https://localhost:{JOPLIN_APP_HOST_PORT}'
+    CMS_API = f"{BASE_URL}/api/graphql"
 else:
-    CMS_API = f"https://{os.getenv('APPNAME','')}.herokuapp.com/api/graphql"
+    BASE_URL = f"https://{os.getenv('APPNAME','')}.herokuapp.com"
```

Comment:
the default is an empty string because otherwise it would be `undefined` and break in a less graceful way?

Reply:
Yes, if we tried to do string interpolation with `None` then everything breaks.